test/test_model.py

```diff
@@ -593,3 +594,30 @@ def test_format_old_version() -> None:
         model.format(max_line_length=88)
 
     model.format(canonicalize=True)
+
+
+def test_diagnose():
```

Comment:
Would you mind adding a test of the `inits` argument? Should be easy enough to test against `log_prob`, actually.